### PR TITLE
features/scaffold-volunteer

### PR DESCRIPTION
**Pull request by chrysalag**
[Link to story in trello](https://trello.com/c/yhqjT4Yx/29-as-a-developer-i-want-to-scaffold-the-volunteer)

- [x] Story under test
- [x] Mobile works
- [x] Seeds created
- [x] Translated

Review by Team
- [x] Reviewed by tnt
- [ ] Reviewed by vok

Review by Coach
- [x] Reviewed by sta

### CODE REVIEW

File: app/controllers/volunteers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  def edit; end
+
+  def create
+    @volunteer = Volunteer.new(volunteer_params)
+
+    respond_to do |format|
+      if @volunteer.save
+        format.html { redirect_to @volunteer, notice: t('volunteer_created') }
+        format.json { render :show, status: :created, location: @volunteer }
```
Comment (kitsane) [inline app/controllers/volunteers_controller.rb]: the `format.json`stuff can be dropped, and the controller simplfied to:
```ruby
if @volunteer.save
  redirect_to @volunteer, notice: t('volunteer_created')
else
  render :new
end
```

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -22,4 +22,31 @@ def week(with_weekend = false)
       [:monday, :tuesday, :wednesday, :thursday, :friday]
     end
   end
+
+  def nationality_name(nationality)
```
Comment (kitsane) [inline app/helpers/application_helper.rb]: Everything that is only used by one model should be moved to the corresponding model. In helpers there should really just be helpers which do something with the view

File: app/views/volunteers/_form.html.slim [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  .row
+    .col-xs-12.col-md-6
+        = f.input :first_name
+
+    .col-xs-12.col-md-6
+        = f.input :last_name
+
+  .row
+    .col-xs-12.col-md-6
+      = f.input :date_of_birth, as: :date, start_year: Date.today.year - 90,
```
Comment (kitsane) [inline app/views/volunteers/_form.html.slim]: You can set default values in the model with an after_initialize callback

File: app/views/volunteers/_form.html.slim [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        end_year: Date.today.year, order: [:day, :month, :year], include_blank: true
```
Comment (kitsane) [inline app/views/volunteers/_form.html.slim]: The order should be configured through the translations i think, but i don't know how to do this with simple_form, please make a little bit of research, i think vok could know

File: test/models/volunteer_test.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  test 'valid factory' do
+    assert @volunteer.valid?
+  end
+
+  test 'voluntary with no required attributes is invalid' do
+    volunteer = Volunteer.new
+    refute volunteer.valid?
+  end
+
+  test 'record still exists after deletion' do
```
Comment (kitsane) [inline test/models/volunteer_test.rb]: Not needed, this is just once to validate paranoid works

File: config/locales/de.yml [hunk 6]
```diff
@@ -74,6 +75,34 @@ de:
...
+        street: Strasse
+        zip: Postleitzahl
+        city: Stadt
+        nationality: Nationalität
+        additional_nationality: Zusätzliche Nationalität
+        phone: Telefonnummer
+        profession: Beruf
+        education: Ausbildung
+        state: Status
+        duration: Dauer
```
Comment (nuritnt) [inline config/locales/de.yml]: Duration of what?

File: app/views/volunteers/show.html.slim [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      tbody
+        tr
+          - single_accompaniment.each do |single|
+            td.text-center
+              - if @volunteer.read_attribute(single)
+                i.glyphicon.glyphicon-ok.text-success
+              - else
+                i.glyphicon.glyphicon-remove.text-danger
+
+  .row
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: outdent this whole block

File: app/views/volunteers/_form.html.slim [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      = f.input :motivation, label: t('your_motivation')
+      = f.input :experience, label: t('your_experience')
+      = f.input :expectations, label: t('your_expectations')
+      = f.input :strengths, label: t('your_strengths')
+      = f.input :skills, label: t('your_skills')
+      = f.input :interests, label: t('your_interests')
+
+  .row
+    .col-xs-12
+      = f.input :state
```
Comment (nuritnt) [inline app/views/volunteers/_form.html.slim]: german translation missing

File: config/locales/de.yml [hunk 9]
```diff
@@ -206,9 +235,20 @@ de:
...
+        gender:
+          female: Weiblich
+          male: Mänlich
+        region:
+          city: Stadt
+          region: Region
+          canton: Kanton
+        duration:
+          short: Kurzfristig
+          long: Längerfristige (min 6 Mt.)
```
Comment (nuritnt) [inline config/locales/de.yml]: Längerfristig

File: app/views/volunteers/show.html.slim [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+            td= t_attr(:profession)
+            td= @volunteer.profession
+          tr
+            td= t_attr(:education)
+            td= @volunteer.education
+          tr
+            td= t_attr(:motivation)
+            td= @volunteer.motivation
+          tr
+            td= t_attr(:experience)
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: german translation missing

File: app/views/volunteers/show.html.slim [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+            td= human_boolean(@volunteer.experience)
+          tr
+            td= t_attr(:expectations)
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: german translation missing

File: app/views/volunteers/show.html.slim [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+            td= @volunteer.expectations
+          tr
+            td= t_attr(:strengths)
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: german translation missing

File: app/views/volunteers/show.html.slim [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+            td= @volunteer.strengths
+          tr
+            td= t_attr(:skills)
+            td= @volunteer.skills
+          tr
+            td= t_attr(:interests)
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: german translation missing

File: app/views/volunteers/show.html.slim [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+            td= t_attr(:experience)
+            td= human_boolean(@volunteer.experience)
+          tr
+            td= t_attr(:expectations)
+            td= @volunteer.expectations
+          tr
+            td= t_attr(:strengths)
+            td= @volunteer.strengths
+          tr
+            td= t_attr(:skills)
```
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: german translation missing

File: app/views/volunteers/show.html.slim (hunk 7 above)
Comment (nuritnt) [inline app/views/volunteers/show.html.slim]: I just ment outdent the group accompaniment block, like the single accompaniment block was before, but its not that important ;)

File: app/views/volunteers/show.html.slim (hunk 10 above)
Comment (chrysalag) [inline app/views/volunteers/show.html.slim]: Added

File: app/views/volunteers/show.html.slim (hunk 11 above)
Comment (chrysalag) [inline app/views/volunteers/show.html.slim]: Added

File: app/views/volunteers/show.html.slim (hunk 12 above)
Comment (chrysalag) [inline app/views/volunteers/show.html.slim]: Added

File: app/views/volunteers/show.html.slim (hunk 14 above)
Comment (chrysalag) [inline app/views/volunteers/show.html.slim]: Added

File: app/views/volunteers/show.html.slim (hunk 13 above)
Comment (chrysalag) [inline app/views/volunteers/show.html.slim]: Added

File: app/views/volunteers/_form.html.slim (hunk 8 above)
Comment (chrysalag) [inline app/views/volunteers/_form.html.slim]: Added

File: app/views/volunteers/_form.html.slim (hunk 4 above)
Comment (chrysalag) [inline app/views/volunteers/_form.html.slim]: I found how this is done. I have to define the order in the local files. For the en do we follow the GB or US format?

File: app/views/volunteers/_form.html.slim (hunk 3 above)
Comment (chrysalag) [inline app/views/volunteers/_form.html.slim]: I did quite some research but I didn't find anything useful. Even at the simple_form documentation examples, they have it as we currently do. If you, or @casaper and @nuritnt have a suggestion or hint about this, please let me know.

File: app/controllers/volunteers_controller.rb [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+  def new
+    @volunteer = Volunteer.new
+  end
+
+  def edit; end
+
+  def create
+    @volunteer = Volunteer.new(volunteer_params)
+    respond_to do |format|
```
Comment (kitsane) [inline app/controllers/volunteers_controller.rb]: drop respond to block

File: app/controllers/volunteers_controller.rb [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+      if @volunteer.save
+        format.html { redirect_to @volunteer, notice: t('volunteer_created') }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
```
Comment (kitsane) [inline app/controllers/volunteers_controller.rb]: drop respond to block

File: app/controllers/volunteers_controller.rb [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+        format.html { redirect_to @volunteer, notice: t('volunteer_updated') }
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
+  def destroy
+    @volunteer.destroy
+    respond_to do |format|
```
Comment (kitsane) [inline app/controllers/volunteers_controller.rb]: drop respond to block

File: app/views/volunteers/_form.html.slim (hunk 3 above)
Comment (kitsane) [inline app/views/volunteers/_form.html.slim]: Don't invest to much time

File: app/views/volunteers/index.html.slim [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+  .col-xs-12
+  table.table.table-striped
+    thead
+      tr
+        th= t_attr(:first_name)
+        th= t_attr(:last_name)
+        th= t_attr(:date_of_birth)
+        th= t_attr(:nationality)
+        th= t_attr(:gender)
+        th Email
```
Comment (kitsane) [inline app/views/volunteers/index.html.slim]: Even if it is the same, also use translation

File: app/views/volunteers/new.html.slim [hunk 19]
```diff
@@ -0,0 +1,11 @@
+.row
+  .col-xs-12
+    h1= t_title(:new)
+
+.row
+  .col-xs-12
+    p= render 'form'
```
Comment (kitsane) [inline app/views/volunteers/new.html.slim]: is the `p` needed?

File: app/views/volunteers/new.html.slim [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+
+.row
+  .col-xs-12
+    p= link_to t('back'), volunteers_path, class: 'btn btn-default'
```
Comment (kitsane) [inline app/views/volunteers/new.html.slim]: is the `p` needed?

File: app/views/volunteers/show.html.slim [hunk 21]
```diff
@@ -0,0 +1,115 @@
+.row
+  .col-xs-12
+    h1
+      = @volunteer.first_name
+      | ,
+      = @volunteer.last_name
```
Comment (kitsane) [inline app/views/volunteers/show.html.slim]: Define a method, called full_name in the model and which returns `#{ last_name }, #{ first_name }` and then just call this here

**Comment by kitsane** [comment]
After the mentioned changes, it looks good.

**Pull request merged**